src/integrationTest/java/uk/gov/hmcts/reform/iacaseapi/integration/AppealReferenceNumberGeneratorIntegrationTest.java

```diff
@@ -0,0 +1,143 @@
+package uk.gov.hmcts.reform.iacaseapi.integration;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertFalse;
+import static uk.gov.hmcts.reform.iacaseapi.integration.stubs.CcdMock.paginationMetadataUrl;
+import static uk.gov.hmcts.reform.iacaseapi.integration.util.TestAsylumCaseBuilder.anAsylumCase;
+import static uk.gov.hmcts.reform.iacaseapi.integration.util.TestFixtures.someListOfCasesIncludingButPriorTo;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.List;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.DirtiesContext;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.AsylumAppealType;
+import uk.gov.hmcts.reform.iacaseapi.domain.service.AppealReferenceNumberGenerator;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.serialization.Serializer;
+import uk.gov.hmcts.reform.iacaseapi.integration.stubs.CcdMock;
+import uk.gov.hmcts.reform.iacaseapi.integration.util.IdamStubbedSpringBootIntegrationTest;
+
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class AppealReferenceNumberGeneratorIntegrationTest extends IdamStubbedSpringBootIntegrationTest {
+
+    @Autowired
+    private AppealReferenceNumberGenerator appealReferenceNumberGenerator;
+
+    @Autowired
+    private Serializer<List> ccdCaseListSerializer;
+
+    private CcdMock givenCcd;
+
+    @Before
+    public void createCcdMock() {
+        givenCcd = new CcdMock(ccdCaseListSerializer);
+    }
+
+    @Test
+    public void generates_revocation_of_protection_appeal_reference_number_when_there_are_no_previous_appeals_in_Ccd() throws JsonProcessingException {
+
+        givenCcd.doesntHaveAnyExistingAppealCases();
+
+        Optional<String> appealReferenceNumber =
+            appealReferenceNumberGenerator
+                .getNextAppealReferenceNumberFor(AsylumAppealType.RP.toString());
+
+        assertThat(appealReferenceNumber.get(), is("RP/50001/2018"));
```

Comment:
tests fail when system clock is moved forward to `2019`, so might need to mock the system's Date value in the test to avoid annual refactor 🙂

Reply:
dates have now been fixed to 2018, but moving to 2019 has not broken them, and a test to cover this scenario has already been included